### PR TITLE
mgr: Centralize PG_STATES to MgrModule

### PR DESCRIPTION
**Pull request by wido**
The Prometheus, Telegraf and Influx module can use this list
of states to send information about Placement Groups to their
endpoint.

Signed-off-by: Wido den Hollander <wido@42on.com>

### CODE REVIEW

**Comment by wido** [comment]
As @jcsp requested this PR which is a WIP.

#22436 and #22434 implement their own list of PG states and by centralizing this we make it dynamic and reduce redundant code.

**Comment by jcsp** [comment]
:+1:

**Comment by wido** [comment]
As the two other PRs have merged I've updated the code.

How does this look?

File: src/pybind/mgr/telegraf/module.py [hunk 1]
```diff
@@ -129,12 +129,7 @@ def get_pg_stats(self):
                     'num_pgs', 'num_objects', 'num_pools']:
             stats[key] = pg_status[key]
 
-        pg_states = ['active', 'peering', 'clean', 'scrubbing', 'undersized',
-                     'backfilling', 'recovering', 'degraded', 'inconsistent',
-                     'remapped', 'backfill_toofull', 'wait_backfill',
-                     'recovery_wait']
-
-        for state in pg_states:
+        for state in PG_STATES:
```
Comment (LenzGr) [inline src/pybind/mgr/telegraf/module.py]: The list of states in PG_STATES is much longer than the one used in the Telegraf module before. Not sure if that has any side effects/consequences?

**Comment by wido** [comment]
@LenzGr Should be no problem for the Telegraf module. You'll just get some more information in your database, bu Telegraf itself won't bother nor the database behind it.

**Comment by wido** [comment]
Fixed!

**Comment by wido** [comment]
jenkins retest this please

**Comment by tchaikov** [comment]
http://pulpito.ceph.com/kchai-2018-07-10_09:44:29-rados-wip-kefu-testing-2018-07-10-1215-distro-basic-mira/

**Pull request merged**